### PR TITLE
feat(idgenerator): Add `IIdGenerator` And Add `MasaApp`

### PR DESCRIPTION
**Pull request by zhenlei520**
# Description

* Add `IIdGenerator` And Add `MasaApp`
* Remove the IsDefault property in Caller

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: src/BuildingBlocks/Data/Masa.BuildingBlocks.Data/IdGenerator/DefaultIdGeneratorFactory.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    public ISequentialGuidGenerator SequentialGuidGenerator => _sequentialGuidGenerator ??=
+        _serviceProvider.GetService<ISequentialGuidGenerator>() ?? throw new Exception($"Unsupported {nameof(SequentialGuidGenerator)}");
+
+    private ISnowflakeGenerator? _snowflakeGenerator;
+
+    public ISnowflakeGenerator SnowflakeGenerator => _snowflakeGenerator ??=
+        _serviceProvider.GetService<ISnowflakeGenerator>() ?? throw new Exception($"Unsupported {nameof(SnowflakeGenerator)}");
+
+    private readonly IServiceProvider _serviceProvider;
```
Comment (doddgu) [inline src/BuildingBlocks/Data/Masa.BuildingBlocks.Data/IdGenerator/DefaultIdGeneratorFactory.cs:23]: Move fileds upon the properties.

File: src/BuildingBlocks/Ddd/Domain/Masa.BuildingBlocks.Ddd.Domain/_Imports.cs [hunk 2]
```diff
@@ -1,8 +1,8 @@
 // Copyright (c) MASA Stack All rights reserved.
 // Licensed under the MIT License. See LICENSE.txt in the project root for license information.
 
-global using Masa.BuildingBlocks.Data.Contracts.DataFiltering;
-global using Masa.BuildingBlocks.Data.Contracts.Paginated;
+global using Masa.BuildingBlocks.Data.Contracts;
+global using Masa.BuildingBlocks.Data.Contracts;
```
Comment (doddgu) [inline src/BuildingBlocks/Ddd/Domain/Masa.BuildingBlocks.Ddd.Domain/_Imports.cs:5]: The using directive for 'Masa.BuildingBlocks.Data.Contracts' appeared previously in this namespace

File: src/BuildingBlocks/Service/Masa.BuildingBlocks.Service.Caller/CallerBase.cs [hunk 3]
```diff
@@ -5,13 +5,16 @@ namespace Masa.BuildingBlocks.Service.Caller;
 
 public abstract class CallerBase
 {
-    public virtual string Name { get; set; } = string.Empty;
+    /// <summary>
+    ///
```
Comment (doddgu) [inline src/BuildingBlocks/Service/Masa.BuildingBlocks.Service.Caller/CallerBase.cs:9]: ?

File: src/Contrib/Service/Masa.Contrib.Service.MinimalAPIs/BackgroundServices/InitializeMasaAppHostedService.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Contrib.Service.MinimalAPIs.BackgroundServices;
+
+public class InitializeMasaAppHostedService : BackgroundService
```
Comment (doddgu) [inline src/Contrib/Service/Masa.Contrib.Service.MinimalAPIs/BackgroundServices/InitializeMasaAppHostedService.cs:6]: Why it's a background service?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Contrib/Service/Masa.Contrib.Service.MinimalAPIs/BackgroundServices/InitializeMasaAppHostedService.cs (hunk 4 above)
Comment (zhenlei520) [reply src/Contrib/Service/Masa.Contrib.Service.MinimalAPIs/BackgroundServices/InitializeMasaAppHostedService.cs]: Processed

File: src/BuildingBlocks/Data/Masa.BuildingBlocks.Data/MasaApp.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.Data;
+
+public class MasaApp
```
Comment (doddgu) [inline src/BuildingBlocks/Data/Masa.BuildingBlocks.Data/MasaApp.cs:6]: Add a 'protected' constructor or the 'static' keyword to the class declaration.

File: src/BuildingBlocks/Data/Masa.BuildingBlocks.Data/IdGenerator/IdGeneratorFactory.cs [hunk 6]
```diff
@@ -6,31 +6,17 @@ namespace Masa.BuildingBlocks.Data;
 public class IdGeneratorFactory
```
Comment (doddgu) [inline src/BuildingBlocks/Data/Masa.BuildingBlocks.Data/IdGenerator/IdGeneratorFactory.cs:6]: Add a 'protected' constructor or the 'static' keyword to the class declaration.

File: src/Contrib/Configuration/Masa.Contrib.Configuration/Extensions/WebApplicationBuilderExtensions.cs [hunk 7]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) MASA Stack All rights reserved.
 // Licensed under the MIT License. See LICENSE.txt in the project root for license information.
 
+using Masa.BuildingBlocks.Data;
```
Comment (doddgu) [inline src/Contrib/Configuration/Masa.Contrib.Configuration/Extensions/WebApplicationBuilderExtensions.cs:4]: Move it to _Imports.

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — APPROVED** [review]
LGTM

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**